lib/src/main/java/io/ably/lib/types/RecoveryKeyContext.java

```diff
@@ -0,0 +1,48 @@
+package io.ably.lib.types;
+
+import com.google.gson.JsonSyntaxException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.ably.lib.util.Log;
+import io.ably.lib.util.Serialisation;
+
+public class RecoveryKeyContext {
+    private static final String TAG = "RecoveryKeyContext";
+
+    private final String connectionKey;
+    private final long msgSerial;
+    private final Map<String, String> channelSerials = new HashMap<>();
+
+    public RecoveryKeyContext(String connectionKey, long msgSerial, Map<String, String> channelSerials) {
+        this.connectionKey = connectionKey;
+        this.msgSerial = msgSerial;
+        this.channelSerials.putAll(channelSerials);
```

Comment:
General principal of immutability. We have marked channelSerials field as immutable, so it shouldn't get influenced by 
changes in external channelserials and should only capture snapshot of serials at the time of creating the instance.

Reply:
Got you! (one more point to rewrite lib in Kotlin) I personally prefer using analog of guava's `ImmutableMap` which just wraps around existing `Map` rather than make `Map`'s copy. But it this particular code probably no harm to copy